src/Components/Search/NewSearch/Mobile/Overlay.tsx

```diff
@@ -34,14 +36,22 @@ interface OverlayProps {
 
 export const Overlay: FC<OverlayProps> = ({ viewer, relay, onClose }) => {
   const { t } = useTranslation()
+  const tracking = useTracking()
   const inputRef = useRef<HTMLInputElement | null>(null)
   const [selectedPill, setSelectedPill] = useState<PillType>(TOP_PILL)
   // TODO: Parse value from url
   const [inputValue, setInputValue] = useState("")
   const disablePills = !shouldStartSearching(inputValue)
+  const [fetchCounter, setFetchCounter] = useState(0)
```

Comment:
Works together with [this piece of code](https://github.com/artsy/force/pull/12567/files#diff-fa4cc42dc68b59e2279b658912183f1658324589a3f28404e27e7255023ba273R60).

This is similar to how it was implemented in the [desktop version of the new search bar](https://github.com/artsy/force/blob/98567f256f5aab0006ebf371144cff93f2e8253e/src/Components/Search/NewSearch/NewSearchBarInput.tsx#L48C10-L48C22).

In short, relay's `refetch` doesn't allow to get response data in the callback function.

```js
relay.refetch(..., error => {
  if (error) { ... }
  
  // here there is no known to me way to get response data
})
```

We've managed to overcome this obstacle by combining `fetchCounter` state and `useEffect` to make sure that tracking event is sent only when component is re-rendered because of new data refetch.

Let me know if you know of a better solution, I would be extremely happy to refactor this one.

Reply:
solved the other way, really there was no need for this counter...